Automattic/jetpack
# [gold] :: Merging paid newsletters PRs.

## Proposed changes:

This PR contains a series of fixes for the Paid Newsletter. We felt it best to combine the PRs below to test all the changes easily. A video of our walkthrough can be found at this link: p81Rsd-1fe-p2

Fixes #29859
Fixes #29710
Fixes #29862

Originally handled #29230 too, but waiting on fix/test.

### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






1. Sandbox `public-api.wordpress.com` and use the production store by disabling the test store in `0-sandbox.php`
2. Start the newsletter signup flow via launchpad by going to `[/setup/newsletter/intro](http://wordpress.com/setup/newsletter/intro)`
3. Once the site is created, **sandbox the site**
4. Create a post, and insert the subscribe block to confirm the “Add payments” button appears on the Subscribe block.
5. Confirm the newsletter access radio buttons are not visible in the Gutenberg panel
6. Go back to the launchpad and connect Stripe
8. Create a newsletter plan
9. Write another post
10. Set the access level to **paid-subscribers**
11. Change post visibility to password protected and verify that an error is shown in the newsletter access panel after entering a password
12. Change post visibility back to public
13. Publish the post for paid subscribers only
14. Open in an incognito window and verify that the user has no access
15. Subscribe using a real card and verify the content is unlocked
16. Subscribe to the same post again but instead select the free subscription option
17. Publish a post for all subscribers, and verify that all subscribers received the newsletter 
18. Publish another paid post, and verify that only the paid subscriber received the post
